.github/workflows/go.yml

```diff
@@ -0,0 +1,36 @@
+name: Go
```

Comment:
I'm fairly inexperienced with Github Actions. Can you explain this file a bit for me?

Reply:
Yep. This is very similar to the way we define CI jobs in Gitlab. The top part tells github actions to only run this on a push or PR to the master branch. The rest of the file checks out the code, gathers the dependencies, builds the binary and runs go tests via `go test ./...`. Please let me know if that doesn't answer all of your questions. Happy to explain further.

Reply:
Makes perfect sense. Thanks for helping me not guess 😉